static/app/views/monitors/components/monitorForm.tsx

```diff
@@ -504,13 +504,22 @@ function MonitorForm({
                   {t('Customize this monitors notification configuration in Alerts')}
                 </AlertLink>
               )}
-              <SentryMemberTeamSelectorField
-                label={t('Notify')}
-                help={t('Send notifications to a member or team.')}
-                name="alertRule.targets"
-                multiple
-                menuPlacement="auto"
-              />
+              <Observer>
+                {() => {
+                  const projectSlug = form.current.getValue('project')?.toString();
+
+                  return (
+                    <SentryMemberTeamSelectorField
+                      label={t('Notify')}
+                      help={t('Send notifications to a member or team.')}
+                      name="alertRule.targets"
+                      memberOfProjectSlug={projectSlug}
+                      multiple
+                      menuPlacement="auto"
+                    />
+                  );
+                }}
+              </Observer>
```

Comment:
```suggestion
              <Observer>
                {() => (
                  <SentryMemberTeamSelectorField
                    label={t('Notify')}
                    help={t('Send notifications to a member or team.')}
                    name="alertRule.targets"
                    memberOfProjectSlug={form.current.getValue('project')?.toString()}
                    multiple
                    menuPlacement="auto"
                  />
                }}
              </Observer>
```

nit: a bit shorter 🤷